.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,24 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
add pull request?
also needs

```
pull_request:
  branches:
    - main
```

Reply:
I think it works even without that, but added anyway.